backend/app/controllers/api/v3/jobs_controller.rb

```diff
@@ -0,0 +1,718 @@
+      def operation_type
+        # Check for manage permissions
+        status, response = check_token_for_permission(Permission.manage_id)
+        render json: response.to_json, status: status.to_sym and return if response[:error]
+
+        # Read category
+        category = Input.text(params[:category]) || ''
+
+        # Read status, default to 'pending'
+        status = case params[:status]
+        when 'error'
+          'error'
+        when 'waiting'
+          'waiting'
+        when 'deferred'
+          'deferred'
+        when 'delayed'
+          'delayed'
+        else
+          'pending'
+        end
+
+        # Read operation type
+        operation_type = Input.text(params[:operation_type]) || ''
+
+        # Get operations for category, type, and status
+        operations = Operation.operations_for_category_type_status(category, operation_type, status)
+
+        render json: { operations: operations }.to_json, status: :ok
+      end
+
+      # Remove an operation from a job
+      #
+      # <b>API Call:</b>
+      #   GET: /api/v3/jobs/<id>/remove/<operation_id>
+      #   {
+      #     token: <token>
+      #   }
+      #
+      # <b>API Return Success:</b>
+      #   STATUS_CODE: 200
+      #   {
+      #     message: "Operation removed"
+      #   }
+      #
+      # @!method remove(id, operation_id, token)
+      # @param id [Int] the id of the job
+      # @param operation_id [Int] the operation_id of the operation
+      # @param token [String] a token
+      def remove
+        # Check for manage permissions
+        status, response = check_token_for_permission(Permission.manage_id)
+        render json: response.to_json, status: status.to_sym and return if response[:error]
+
+        # Get job
+        id = Input.int(params[:id])
+        job = Job.find_by(id: id)
+        render json: { error: "Job not found" }.to_json, status: :not_found and return if !job
+
+        # Check job status
+        render json: { error: "Job must be not started" }.to_json, status: :unauthorized and return if job.pc != -1
+
+        # Get operation (and verify that it is in the job)
+        operation_id = Input.int(params[:operation_id])
+        operation = Operation.operation_from_job(operation_id, job.id)
+        render json: { error: "Operation not found" }.to_json, status: :not_found and return if !operation
+
+        # Check operation status
+        render json: { error: "Operation must be scheduled" }.to_json, status: :unauthorized and return if operation.status != 'scheduled'
+
+        # remove operation from job
+        JobAssociation.remove_operation_from_job(operation_id, job.id)
+
+        # update the operation status
+        Operation.set_status_for_ids('pending', [operation_id])
+
+        # NOTE: may want to remove the job if there are no operations left in the job
+
+        render json: { message: "Operation removed" }.to_json, status: :ok
+      end
+
+      private
+
+      def new_job_assignment_log
```

Comment:
I can move this.  Personally I like breaking convention a bit here, because the end-user is really manipulating the "Job" (i.e., "assign this job to person x") and the data just happens to be saved in the job_assignment_logs table - but I'm happy to move it.

Reply:
done (moved).